packages/api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -221,66 +221,73 @@ async function getOrderNumberFromSession(
 // Returns the regionalized orderForm
 const getOrderForm = async (
   id: string,
-  session: Maybe<IStoreSession> | undefined,
   { clients: { commerce } }: Context
 ) => {
   const orderForm = await commerce.checkout.orderForm({
     id,
   })
 
+  return orderForm
+}
+
+const updateOrderFormShippingData = async (
+  orderForm: OrderForm,
+  session: Maybe<IStoreSession> | undefined,
+  { clients: { commerce } }: Context)=>
+  {
   // Stores that are not yet providing the session while validating the cart
   // should not be able to update the shipping data
   //
   // This was causing errors while validating regionalizated carts
   // because the following code was trying to change the shippingData to an undefined address/session
-  if (!session) {
-    return orderForm
-  }
-
-  const { updateShipping, addressChanged } = shouldUpdateShippingData(
-    orderForm,
-    session
-  )
-
-  if (updateShipping) {
-    // Check if the orderForm address matches the one from the session
-    const oldAddress = getAddressOrderForm(orderForm, session, addressChanged)
-
-    const address = oldAddress ? oldAddress : createNewAddress(session)
-
-    const selectedAddresses = address as SelectedAddress[]
 
-    const hasDeliveryWindow = session.deliveryMode?.deliveryWindow
-      ? true
-      : false
-
-    if (hasDeliveryWindow) {
-      // if you have a Delivery Window you have to first get the delivery window to set the desired after
-      await commerce.checkout.shippingData(
+    if (!session) {
+      return orderForm
```

Comment:
If return isn't used, just return empty
```suggestion
      return
```